src/Compilers/CSharp/Portable/Compilation/MethodBodySemanticModel.cs

```diff
@@ -28,15 +48,14 @@ internal sealed class MethodBodySemanticModel : MemberSemanticModel
         /// <summary>
         /// Creates a SemanticModel for the method.
         /// </summary>
-        internal static MethodBodySemanticModel Create(SyntaxTreeSemanticModel containingSemanticModel, MethodSymbol owner, ExecutableCodeBinder executableCodeBinder,
-                                                       CSharpSyntaxNode syntax, BoundNode boundNode = null, NullableWalker.SnapshotManager snapshotManager = null)
+        internal static MethodBodySemanticModel Create(SyntaxTreeSemanticModel containingSemanticModel, MethodSymbol owner, InitialState initialState)
         {
             Debug.Assert(containingSemanticModel != null);
-            var result = new MethodBodySemanticModel(owner, executableCodeBinder, syntax, containingSemanticModel);
+            var result = new MethodBodySemanticModel(owner, initialState.Binder, initialState.Syntax, containingSemanticModel);
 
-            if (boundNode != null)
+            if (initialState.Body != null)
             {
-                result.UnguardedAddBoundTreeForStandaloneSyntax(syntax, boundNode, snapshotManager);
+                result.UnguardedAddBoundTreeForStandaloneSyntax(initialState.Syntax, initialState.Body, initialState.SnapshotManager, initialState.AnalyzedVariableDeclarations is { } variables ? GetUpdatedLocalSymbols(variables) : null);
```

Comment:
>{ } [](start = 179, length = 3)

`object` preferred over `{ }`

Reply:
I can't. `{ }` preserves the existing type, `object` does not.

---
In reply to: [305055503](https://github.com/dotnet/roslyn/pull/37322#discussion_r305055503) [](ancestors = 305055503)